src/Build.UnitTests/BackEnd/BuildManager_Tests.cs

```diff
@@ -944,6 +944,68 @@ public void SimpleBuildWithFailure()
             _logger.AssertLogContains("[errormessage]");
         }
 
+        [Fact]
+        public void DeferredMessageShouldGetLogged()
```

Comment:
nit:
```suggestion
        public void DeferredMessageShouldBeLogged()
```
(Slightly more formal, and it matches the shouldly asserts better)